redaxo/src/core/lib/packages/addons/addon.php

```diff
@@ -344,19 +344,18 @@ public static function initialize($dbExists = true)
     /**
      * Filters packages by the given method.
      *
-     * @param rex_package[] $packages Array of packages
-     * @param string        $method   A rex_package method
-     *
-     * @return rex_package[]
-     *
      * @template T of rex_package
-     * @psalm-param T[] $packages
-     * @psalm-return T[]
+     * @param array<string, T> $packages Array of packages
+     * @param string $method   A rex_package method
+     * @return array<string, T>
      */
     private static function filterPackages(array $packages, $method)
     {
-        return array_filter($packages, static function (rex_package $package) use ($method) {
-            return $package->$method();
+        return array_filter($packages, static function (rex_package $package) use ($method): bool {
+            $return = $package->$method();
+            assert(is_bool($return));
```

Comment:
ich bin mittlerweile kein fan mehr von den `assert` da sie zur laufzeit nicht laufen und damit zur laufzeit errors/warnings passieren können, die in DEV aufgrund des `assert` ignoriert werden.

und in DEV funktioinerts nur wenn die php.ini entsprechend eingestellt ist.. daher würd ich hier eine RuntimeException o.ä. werfen

Reply:
An der konkreten Stelle hier wird es übrigens auch auf prod knallen, da ich den Return-Type der Callback-Function auf `bool` gesetzt habe.